src/compiler/transformers/transform-utils.ts

```diff
@@ -945,3 +945,97 @@ export const retrieveTsDecorators = (node: ts.Node): ReadonlyArray<ts.Decorator>
 export const retrieveTsModifiers = (node: ts.Node): ReadonlyArray<ts.Modifier> | undefined => {
   return ts.canHaveModifiers(node) ? ts.getModifiers(node) : undefined;
 };
+
+/**
+ * Helper util for updating the constructor on a class declaration AST node.
+ *
+ * @param classNode the class node whose constructor will be updated
+ * @param classMembers a list of class members for that class
+ * @param statements a list of statements which should be added to the
+ * constructor
+ * @returns a list of updated class elements
+ */
+export const updateConstructor = (
+  classNode: ts.ClassDeclaration,
+  classMembers: ts.ClassElement[],
+  statements: ts.Statement[],
+): ts.ClassElement[] => {
+  const constructorIndex = classMembers.findIndex((m) => m.kind === ts.SyntaxKind.Constructor);
+  const constructorMethod = classMembers[constructorIndex];
+
+  if (constructorIndex >= 0 && ts.isConstructorDeclaration(constructorMethod)) {
+    const constructorBodyStatements: ts.NodeArray<ts.Statement> =
+      constructorMethod.body?.statements ?? ts.factory.createNodeArray();
+    const hasSuper = constructorBodyStatements.some((s) => s.kind === ts.SyntaxKind.SuperKeyword);
+
+    if (!hasSuper && needsSuper(classNode)) {
+      // if there is no super and it needs one the statements comprising the
+      // body of the constructor should be:
+      //
+      // 1. the `super()` call
+      // 2. the new statements we've created to initialize fields
+      // 3. the statements currently comprising the body of the constructor
+      statements = [createConstructorBodyWithSuper(), ...statements, ...constructorBodyStatements];
+    } else {
+      // if no super is needed then the body of the constructor should be:
+      //
+      // 1. the new statements we've created
+      // 2. the statements currently comprising the body of the constructor
+      statements = [...statements, ...constructorBodyStatements];
+    }
+
+    classMembers[constructorIndex] = ts.factory.updateConstructorDeclaration(
+      constructorMethod,
+      retrieveTsModifiers(constructorMethod),
+      constructorMethod.parameters,
+      ts.factory.updateBlock(constructorMethod?.body ?? ts.factory.createBlock([]), statements),
+    );
+  } else {
+    // we don't seem to have a constructor, so let's create one and stick it
+    // into the array of class elements
+    if (needsSuper(classNode)) {
+      statements = [createConstructorBodyWithSuper(), ...statements];
+    }
+
+    // add the new constructor to the class members, putting it at the
+    // beginning
+    classMembers.unshift(
+      ts.factory.createConstructorDeclaration(undefined, [], ts.factory.createBlock(statements, true)),
+    );
```

Comment:
Verifying my understanding - in moving this function from `convert-decorators` to here, is this the only block of code (other than comments) that changed? 

Comparing this against - https://github.com/ionic-team/stencil/pull/4741/files#diff-84d52fb230ffdf800b6b3debb4f554136efd13683b396c03d1dd8a129f81a602L464-L468

```diff
-    classMembers = [
-      ts.factory.createConstructorDeclaration(undefined, [], ts.factory.createBlock(statements, true)),
-      ...classMembers,
-    ];
+    classMembers.unshift(
+     ts.factory.createConstructorDeclaration(undefined, [], ts.factory.createBlock(statements, true)),
+    );
```

Reply:
yep! I made that change in order to ensure that the `classMembers` array that's passed in gets modified, since that aligns with how the modification of the constructor for the native constructor transformers was being done